### ISSUE TITLE
Can't use the CMS without Campaigns Admin module

### ISSUE THREAD

**Issue by baukezwaan** [silverstripe/silverstripe-cms#2242]
SilverStripe 4.2.1

I would like to remove the campaigns functionality from the CMS. Now I could hide it by

`CMSMenu::remove_menu_class(CampaignAmdin::class);`

but I would like to completely remove it via compose.json, by:

```
“replace”: {
  “silverstripe/campaign-admin”: “*”
},
```

Problem now is that I get an PHP-error:

`[Emergency] Uncaught Error: Class 'SilverStripe\CampaignAdmin\AddToCampaignHandler_FormAction' not found`

because in [SiteTree.php:2267](https://github.com/silverstripe/silverstripe-cms/blob/4/code/Model/SiteTree.php#L2267) Campaign buttons are trying to being added to the Pages. 

Could it be fixed by checking if the CampaignAdmin class exists:

`        if (($isPublished || $isOnDraft) && $canPublish ) {`
to
`        if (($isPublished || $isOnDraft) && $canPublish && class_exists('CampaignAdmin')) {`

**Comment by robbieaverill** [comment]
Yeah we shouldn’t have this kind of coupling. Ideally that code lives in the campaign admin module, but if you’re keen to make a pull request we would accept one to implement your suggested change :-)

**Comment by robbieaverill** [comment]
We would also need to check for other places where it’s referenced

**Comment by robbieaverill** [comment]
Oh, I’m sorry! It’s a direct dependency of this module so it wouldn’t make sense to make it conditional right now. You could make a PR anyway and see what the core team think

**Comment by maxime-rainville** [comment]
I'm with @baukezwaan, campaign-admin and CMS could be decoupled so developers who don't want the campaign functionality, can remove it from their CMS project.

**Comment by kinglozzer** [comment]
Related: https://github.com/silverstripe/silverstripe-campaign-admin/issues/1
